tools/src/main/java/org/apache/kafka/trogdor/workload/Histogram.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.trogdor.workload;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * A histogram that can easily find the average, median etc of a large number of samples in a
+ * restricted domain.
+ */
+public class Histogram {
+    private final int[] counts;
+
+    private final Logger log = LoggerFactory.getLogger(Histogram.class);
+
+    public Histogram(int maxValue) {
+        this.counts = new int[maxValue + 1];
+    }
+
+    public void add(long value) {
```

Comment:
Since this looks like a utility class that could be reused, it would be better for `add(long)` to store longs rather than truncate silently. If we only want to store ints, it may be better to have a static `toInt` method that does the conversion to `int` in the caller and only provide `add(int)` in `Histogram`.

Reply:
Since Histogram has a fixed number of buckets, things that are greater than the maximum number of buckets go in the last bucket.  This is consistent with that semantic, so I'm not sure if it's an issue here.

In general, latencies more than 2**31 milliseconds should really never happen (it indicates some other error, like someone changing the clock perhaps).  I can see that perhaps having the method which takes a long could be misleading, because it might make people believe they could store longs in this class (they cannot: arrays can't have more than Integer.MAX_VALUE elements).  Perhaps a comment here would help?

Reply:
Yes, a comment would be useful, thanks.